### PR TITLE
fix: compatibility check not working in bash

### PR DESCRIPTION
**Pull request by gbtec-markusmeier**
fixes https://github.com/wfxr/forgit/issues/256
fixes https://github.com/wfxr/forgit/issues/257

## Check list

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation change

## Test environment

- Shell
    - [x] bash
    - [x] zsh
    - [ ] fish
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Others:

Fixes:
* https://github.com/wfxr/forgit/issues/256
* https://github.com/wfxr/forgit/issues/257

### CODE REVIEW

File: forgit.plugin.zsh [hunk 1]
```diff
@@ -21,8 +21,9 @@ FORGIT="$INSTALL_DIR/bin/git-forgit"
 # backwards compatibility:
 # export all user-defined FORGIT variables to make them available in git-forgit
 unexported_vars=0
+[[ -n "$BASH_VERSION" ]] && set -o posix
```
Comment (carlfriedrich) [inline forgit.plugin.zsh:24]: Maybe it would be good to add a describing comment here why we're calling `set -o posix`.

**Review by carlfriedrich — CHANGES_REQUESTED** [review]
@gbtec-markusmeier Thanks a lot for your bug reports and this fix. Do I understand it correctly that this solution only fixes the situation for `bash`? Do we need a different behavior for `zsh`?

**Comment by gbtec-markusmeier** [comment]
> @gbtec-markusmeier Thanks a lot for your bug reports and this fix. Do I understand it correctly that this solution only fixes the situation for `bash`? Do we need a different behavior for `zsh`?

Zsh works fine, we don't have that problem there: zsh reports only variables when calling `set`.

**Review by carlfriedrich — APPROVED** [review]
@gbtec-markusmeier Great, thanks a lot. Since it works for me as well in both shells, I will directly merge this. In fish this hasn't been reported as a problem so far.

**Pull request merged**